source/mongodb/tables/role-configuration.rst

```diff
@@ -36,82 +41,159 @@
 
    * - | ``apply_when``
        | Expression
-     - An :ref:`expression <expressions>` that evaluates to ``true`` when
-       this role applies to a user for a specific document.
+     - An :ref:`expression <expressions>` that evaluates to true when
+       this :ref:`role <roles>` applies to a user.
+
+       When Device Sync (Flexible Mode) is not enabled, App Services assigns a
+       role on a per-document basis. When Device Sync (Flexible Mode) is
+       enabled, App Services assigns roles on a per-collection, per-session
+       basis -- that is, for each synced collection when a client opens a sync
+       connection.
+
+       To assign a role, App Services evaluates the ``apply_when`` of each
+       relevant role until one evaluates to true. Relevant roles are any roles
+       specified for the given collection plus the default role(s). App Services
+       evaluates collection-specific roles in the order that you specify them in
+       your configuration. If no collection-specific role applies, App Services
```

Comment:
I reworked this completely - realizing we don't fall through to default roles, but it's either or. Check it out and let me know.

Reply:
Looks better! I know I'm probably being overly pedantic here, but I think _technically_
```
Potential roles are any roles
specified for the given collection or, if no roles are specified for the
given collection, the default role(s)
```
would not apply to the example I mentioned above (ie, even though there are no roles in the namespace rule, the default roles do not apply because the namespace rule exists). 

If we're trying to avoid referencing "rule" in the snippet here - I get it, and would be fine with the existing language, but maybe we could add a note somewhere about this odd edge case / behavior?

Reply:
I see. We haven't really defined rule as the file that you put roles in. Let's see if I can continue to avoid it.